packages/contracts-bedrock/contracts/test/OptimismPortal.t.sol

```diff
@@ -1083,3 +1083,208 @@ contract OptimismPortalUpgradeable_Test is Portal_Initializer {
+    Types.WithdrawalTransaction _defaultTx;
+    uint256 _proposedOutputIndex;
+    uint256 _proposedBlockNumber;
+    bytes32 _stateRoot;
+    bytes32 _storageRoot;
+    bytes32 _outputRoot;
+    bytes32 _withdrawalHash;
+    bytes[] _withdrawalProof;
+    Types.OutputRootProof internal _outputRootProof;
+
+    // The receiver of the withdrawal message.
+    IMessageReceiver internal receiver;
+
+    function test_finalizeWithdrawalTransaction_minGasLimitCheckGap_succeeds() public {
+        // Prepare the test with a withdrawal transaction that has a minimum gas limit of
+        // 22_203. This is the exact amount of gas required to execute `receiveMessage` in
+        // the `MessageReceiverCheckGap` contract.
+        uint256 minGasLimit = 22_203;
+        _prep(minGasLimit, IMessageReceiver(address(new MessageReceiverCheckGap())));
+
+        // Prove the withdrawal transaction
+        op.proveWithdrawalTransaction(
+            _defaultTx,
+            _proposedOutputIndex,
+            _outputRootProof,
+            _withdrawalProof
+        );
+
+        // Warp past the finalization period
+        vm.warp(block.timestamp + op.FINALIZATION_PERIOD_SECONDS() + 1);
+
+        // Loop through a range of gas values, and ensure that the external call *always* receives
+        // at LEAST the minimum gas limit each time the call to `finalizeWithdrawalTransaction`
+        // succeeds.
+        //
+        // The range [75_483, 85_483] was chosen because it's a range that includes the minimum amount
+        // of gas required to execute this `finalizeWithdrawalTransaction` call, 80_483, +/- 10_000.
+        for (uint256 gas = 70_483; gas <= 90_483; gas++) {
+            // Take a state snapshot.
+            uint256 snapshotId = vm.snapshot();
+
+            // Attempt to finalize the withdrawal.
+            (bool success, bytes memory returndata) = address(op).call{ gas: gas }(
+                abi.encodeWithSelector(op.finalizeWithdrawalTransaction.selector, _defaultTx)
+            );
+
+            // If the call was successful, then the gas supplied to the external call MUST be greater than the
+            // minimum gas limit. Otherwise, the call MUST revert with the expected error message.
+            if (success) {
+                // This assertion doubles as a check that the sub-call succeeded AND that the gas
+                // supplied was greater than or equal to the `minGasLimit`.
+                assertTrue(receiver.receivedGas() >= minGasLimit);
+            } else {
+                assertEq(
+                    returndata,
+                    abi.encodeWithSignature(
+                        "Error(string)",
+                        "OptimismPortal: insufficient gas to finalize withdrawal"
+                    )
+                );
+            }
+
+            // Revert back to the snapshot state (before the withdrawal was finalized).
+            assertTrue(vm.revertTo(snapshotId));
```

Comment:
If this resets hot storage slots thats amazing

Reply:
It does 😄 

The issue still remains with any storage slots that were warm before the snapshot due to deployment / proving happening in the same transaction, though. This means the slot for the `withdrawalHash` within the `provenWithdrawals` mapping is be warm, etc.

<img width="672" alt="Screenshot 2023-02-23 at 12 03 59 AM" src="https://user-images.githubusercontent.com/8406232/220825397-e764c145-cb9e-4588-b01d-2b9e485f84ac.png">